src/codegen/x86_64/arch_x86_64.c

```diff
@@ -186,13 +186,36 @@ enum Instruction {
   I_COUNT
 };
 
+enum RegSize {
+  r128,
+  r64,
+  r32,
+  r16,
+  r8,
+};
+
+/// Return the corresponding RegSize enum value to the given amount of
+/// bytes (smallest fit). ICE if can not contain.
+static enum RegSize regsize_from_bytes(u64 bytes) {
+  switch (bytes) {
+  case 1: return r8;
+  case 2: return r16;
+  case 4: return r32;
+  case 8: return r64;
+  case 16: return r128;
+  default:
+    ICE("Byte size can not be converted into register size on x86_64: %U", bytes);
+    break;
+  }
+}
+
 enum InstructionOperands_x86_64 {
   IMMEDIATE, ///< int64_t imm
   MEMORY,    ///< Reg reg, int64_t offset
   REGISTER,  ///< Reg reg
   NAME,      ///< const char* name
 
-  IMMEDIATE_TO_REGISTER, ///< int64_t imm, Reg dest
+  IMMEDIATE_TO_REGISTER, ///< int64_t imm, Reg dest, RegSize size
   IMMEDIATE_TO_MEMORY,   ///< int64_t imm, Reg address, int64_t offset
   MEMORY_TO_REGISTER,    ///< Reg address, int64_t offset, Reg dest
```

Comment:
That's definitely true; I wrote the docs to early and they no longer apply; a tale as old as time haha. I'll be sure to get that fixed up, since this is already a place that bugs are likely to occur in; the last thing we need is faulty/incorrect docs